sdk/keyvault/azure-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/SecretAsyncClient.java

```diff
@@ -220,58 +222,58 @@
      * operation is applicable to any secret stored in Azure Key Vault. This operation requires the
      * {@code secrets/get} permission.
      *
      * <p>The list operations {@link SecretAsyncClient#listSecrets()} and {@link
-     * SecretAsyncClient#listSecretVersions(String)} return the {@link Flux} containing {@link SecretBase base secret}
-     * as output. This operation can then be used to get the full secret with its value from {@code secretBase}.</p>
+     * SecretAsyncClient#listSecretVersions(String)} return the {@link Flux} containing {@link SecretProperties base secret}
+     * as output. This operation can then be used to get the full secret with its value from {@code secretProperties}.</p>
      *
      * <p><strong>Code Samples</strong></p>
-     * {@codesnippet com.azure.keyvault.secrets.secretclient.getSecretWithResponse#secretBase}
+     * {@codesnippet com.azure.keyvault.secrets.secretclient.getSecretWithResponse#secretProperties}
      *
-     * @param secretBase The {@link SecretBase base secret} secret base holding attributes of the secret being
+     * @param secretProperties The {@link SecretProperties base secret} secret base holding attributes of the secret being
      *     requested.
      * @return A {@link Response} whose {@link Response#getValue() value} contains the requested {@link Secret secret}.
-     * @throws ResourceNotFoundException when a secret with {@link SecretBase#getName() name} and {@link
-     *     SecretBase#getVersion() version} doesn't exist in the key vault.
-     * @throws HttpRequestException if {@link SecretBase#getName()}  name} or {@link SecretBase#getVersion() version} is empty
+     * @throws ResourceNotFoundException when a secret with {@link SecretProperties#getName() name} and {@link
+     *     SecretProperties#getVersion() version} doesn't exist in the key vault.
+     * @throws HttpRequestException if {@link SecretProperties#getName()}  name} or {@link SecretProperties#getVersion() version} is empty
      *     string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<Secret>> getSecretWithResponse(SecretBase secretBase) {
-        return withContext(context -> getSecretWithResponse(secretBase, context));
+    public Mono<Response<Secret>> getSecretWithResponse(SecretProperties secretProperties) {
+        return withContext(context -> getSecretWithResponse(secretProperties, context));
     }
 
-    Mono<Response<Secret>> getSecretWithResponse(SecretBase secretBase, Context context) {
-        Objects.requireNonNull(secretBase, "The Secret Base parameter cannot be null.");
-        return getSecretWithResponse(secretBase.getName(), secretBase.getVersion() == null ? "" : secretBase.getVersion(),
+    Mono<Response<Secret>> getSecretWithResponse(SecretProperties secretProperties, Context context) {
+        Objects.requireNonNull(secretProperties, "The Secret Base parameter cannot be null.");
```

Comment:
"The ~~Secret Base~~  Secret Properties parameter ..."

Reply:
fixed